test/lib/jdk/test/lib/artifacts/ArtifactManager.java

```diff
@@ -24,7 +24,9 @@
 package jdk.test.lib.artifacts;
 
 import java.nio.file.Path;
+import java.util.Map;
 
 public interface ArtifactManager {
     public Path resolve(Artifact artifact) throws ArtifactResolverException;
+    Path resolve(String name, Map<String, Object> artifactDescription, boolean unpack) throws ArtifactResolverException;
```

Comment:
For consistency, `public` should be added, similar to the existing interface method.

Also, ArtifactManager is a public interface and some JDK developers may have written custom classes of this interface. If we add a new interface method, we may break such custom classes.

For compatibility, the new method should be declared with a default implementation:

```
default public Path resolve(String name, Map<String, Object> artifactDescription, boolean unpack) throws ArtifactResolverException {
    throw new ArtifactResolverException("not implemented");
}
```

Note that it throws the exception instead of returning null. This will be consistent with the existing interface method, which throws ArtifactResolverException in case of any error.

Otherwise, the caller of the new method would need to handle both the null return and the ArtifactResolverException, which would be too clumsy.